datastore/admin/datastore_admin_index_list.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package samples
+
+// [START datastore_admin_index_list]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	admin "cloud.google.com/go/datastore/admin/apiv1"
+	"google.golang.org/api/iterator"
+	adminpb "google.golang.org/genproto/googleapis/datastore/admin/v1"
+)
+
+// indexList lists the indexes.
+func indexList(w io.Writer, projectID string) ([]*adminpb.Index, error) {
+	// projectID := "my-project-id"
+	ctx := context.Background()
+	client, err := admin.NewDatastoreAdminClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("admin.NewDatastoreAdminClient: %v", err)
+	}
+	defer client.Close()
+
+	req := &adminpb.ListIndexesRequest{
+		ProjectId: projectID}
+	it := client.ListIndexes(ctx, req)
+	var indexArr []*adminpb.Index
```

Comment:
```suggestion
	var indices []*adminpb.Index
```

Technically, this is a slice, not an array. A slice is dynamically sized and stores data in arrays under the hood. https://blog.golang.org/slices-intro